### ISSUE TITLE
Converting 'record' to 'class' throws ambiguous errors

### ISSUE THREAD

**Issue by arsinclair** [dotnet/vscode-csharp#4313]
## Issue Description ##
I have the following bit of code: https://gist.github.com/arsinclair/76d6842ae68d83092354a8bfb2a30093

Once I convert it to class, it produces various errors (see screenshots below). The conversion is done by replacing the word `record` with the word `class`.

Errors distributed across the file:
![Error distribution](https://user-images.githubusercontent.com/2878904/102864280-841aaf80-443c-11eb-867e-f8180bae8566.png)

The error descriptions and their location:
![Error descriptions](https://user-images.githubusercontent.com/2878904/102864309-97c61600-443c-11eb-8161-86b81be812bc.png)

## Reproduction steps ##
The reproduction is unclear: restarting vscode has exited this weird state. Though, if I repeat my initial steps, I can see that all these errors appear when I haven't yet finished writing the word `class`, but once the word is there: they disappear. So initially, the case was that once the word has been completed, the errors didn't go away.

## Expected Behavior ##
These errors don't make any sense. If I'm doing something wrong, the errors should tell me what's wrong.

### OmniSharp log ###
<details>Starting OmniSharp server at 22/12/2020, 09:50:37
    Target: /home/jrm/GIT/contentful-extractor/contentful-json-reducer

OmniSharp server started.
    Path: /home/jrm/.vscode/extensions/ms-dotnettools.csharp-1.23.8/.omnisharp/1.37.5/run
    PID: 9674

[info]: OmniSharp.Stdio.Host
        Starting OmniSharp on debian 10.0 (x64)
[info]: OmniSharp.Services.DotNetCliService
        DotNetPath set to dotnet
[info]: OmniSharp.MSBuild.Discovery.MSBuildLocator
        Located 1 MSBuild instance(s)
            1: StandAlone 16.8.0 - "/home/jrm/.vscode/extensions/ms-dotnettools.csharp-1.23.8/.omnisharp/1.37.5/omnisharp/.msbuild/Current/Bin"
[info]: OmniSharp.MSBuild.Discovery.MSBuildLocator
        MSBUILD_EXE_PATH environment variable set to '/home/jrm/.vscode/extensions/ms-dotnettools.csharp-1.23.8/.omnisharp/1.37.5/omnisharp/.msbuild/Current/Bin/MSBuild.exe'
[info]: OmniSharp.MSBuild.Discovery.MSBuildLocator
        Registered MSBuild instance: StandAlone 16.8.0 - "/home/jrm/.vscode/extensions/ms-dotnettools.csharp-1.23.8/.omnisharp/1.37.5/omnisharp/.msbuild/Current/Bin"
            CscToolExe = csc.exe
            MSBuildToolsPath = /home/jrm/.vscode/extensions/ms-dotnettools.csharp-1.23.8/.omnisharp/1.37.5/omnisharp/.msbuild/Current/Bin
            CscToolPath = /home/jrm/.vscode/extensions/ms-dotnettools.csharp-1.23.8/.omnisharp/1.37.5/omnisharp/.msbuild/Current/Bin/Roslyn
            BypassFrameworkInstallChecks = true
            MSBuildExtensionsPath = /home/jrm/.vscode/extensions/ms-dotnettools.csharp-1.23.8/.omnisharp/1.37.5/omnisharp/.msbuild
[info]: OmniSharp.Cake.CakeProjectSystem
        Detecting Cake files in '/home/jrm/GIT/contentful-extractor/contentful-json-reducer'.
[info]: OmniSharp.Cake.CakeProjectSystem
        Could not find any Cake files
[info]: OmniSharp.MSBuild.ProjectSystem
        No solution files found in '/home/jrm/GIT/contentful-extractor/contentful-json-reducer'
[info]: OmniSharp.MSBuild.ProjectManager
        Queue project update for '/home/jrm/GIT/contentful-extractor/contentful-json-reducer/contentful-json-reducer.csproj'
[info]: OmniSharp.Script.ScriptProjectSystem
        Detecting CSX files in '/home/jrm/GIT/contentful-extractor/contentful-json-reducer'.
[info]: OmniSharp.Script.ScriptProjectSystem
        Could not find any CSX files
[info]: OmniSharp.WorkspaceInitializer
        Invoking Workspace Options Provider: OmniSharp.Roslyn.CSharp.Services.CSharpFormattingWorkspaceOptionsProvider, Order: 0
[info]: OmniSharp.MSBuild.ProjectManager
        Loading project: /home/jrm/GIT/contentful-extractor/contentful-json-reducer/contentful-json-reducer.csproj
[info]: OmniSharp.WorkspaceInitializer
        Invoking Workspace Options Provider: OmniSharp.Roslyn.CSharp.Services.Completion.CompletionOptionsProvider, Order: 0
[info]: OmniSharp.WorkspaceInitializer
        Invoking Workspace Options Provider: OmniSharp.Roslyn.CSharp.Services.RenameWorkspaceOptionsProvider, Order: 100
[info]: OmniSharp.WorkspaceInitializer
        Invoking Workspace Options Provider: OmniSharp.Roslyn.CSharp.Services.ImplementTypeWorkspaceOptionsProvider, Order: 110
[info]: OmniSharp.WorkspaceInitializer
        Invoking Workspace Options Provider: OmniSharp.Roslyn.CSharp.Services.BlockStructureWorkspaceOptionsProvider, Order: 140
[info]: OmniSharp.WorkspaceInitializer
        Configuration finished.
[info]: OmniSharp.Stdio.Host
        Omnisharp server running using Stdio at location '/home/jrm/GIT/contentful-extractor/contentful-json-reducer' on host 9451.
[info]: OmniSharp.MSBuild.ProjectManager
        Successfully loaded project file '/home/jrm/GIT/contentful-extractor/contentful-json-reducer/contentful-json-reducer.csproj'.
[info]: OmniSharp.MSBuild.ProjectManager
        Adding project '/home/jrm/GIT/contentful-extractor/contentful-json-reducer/contentful-json-reducer.csproj'
[info]: OmniSharp.MSBuild.ProjectManager
        Update project: contentful-json-reducer
</details>

### C# log ###
<details>Empty</details>

## Environment information ##

**VSCode version**: 1.52.1
**C# Extension**: 1.23.8

<details><summary>Mono Information</summary>
        OmniSharp using built-in mono</details>
<details><summary>Dotnet Information</summary>
.NET SDK (reflecting any global.json):
 Version:   5.0.101
 Commit:    d05174dc5a

Runtime Environment:
 OS Name:     debian
 OS Version:  10
 OS Platform: Linux
 RID:         debian.10-x64
 Base Path:   /usr/share/dotnet/sdk/5.0.101/

Host (useful for support):
  Version: 5.0.1
  Commit:  b02e13abab

.NET SDKs installed:
  3.1.404 [/usr/share/dotnet/sdk]
  5.0.101 [/usr/share/dotnet/sdk]

.NET runtimes installed:
  Microsoft.AspNetCore.App 3.1.10 [/usr/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 5.0.1 [/usr/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 3.1.10 [/usr/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 5.0.1 [/usr/share/dotnet/shared/Microsoft.NETCore.App]

To install additional .NET runtimes or SDKs:
  https://aka.ms/dotnet-download
</details>
<details><summary>Visual Studio Code Extensions</summary>

|Extension|Author|Version|
|---|---|---|
|blazorsnippets|ScottSauber|1.3.0|
|csharp|ms-dotnettools|1.23.8|
|es7-react-js-snippets|dsznajder|3.1.0|
|jbockle-format-files|jbockle|3.0.0|
|theme-dracula|dracula-theme|2.22.3|
|transformer|dakara|1.11.0|
|vscode-css-formatter|aeschli|1.0.1|
|vscode-peacock|johnpapa|3.9.1|;

</details>

**Comment by JoeRobich** [comment]
A lot has changed with O# since this issue was opened. If you are still having problems, please open a new issue at https://github.com/omnisharp/omnisharp-roslyn/issues

**Comment by arsinclair** [comment]
Thank you for basically negating the time I've spent on submitting this bug request. But I'm not going to do it again.

Given the instability of the ecosystem and lack of attention towards the DX from Microsoft for years, a while ago I've completely dropped .net and shifted gears to TS. Not planning to come back.

**Comment by JoeRobich** [comment]
@arsinclair O# has a long standing performance issue where it sends the entire text buffer to the server on each request. At different times we have tried to address this. You report is consistent with a bug that was introduced during one of these times. 

![Image](https://github.com/user-attachments/assets/30ffac25-acbf-4c61-913c-9e6cb628c664)

The issue was that the editor and O# would have different views of the text buffer.  The error above shows the problem pretty clearly. The server believes the source file contains `rlassrecord`. The problematic change was reverted and a new build published but it appears you were affected by it.

I am sorry this issue wasn't responded to or closed out when the issue was resolved.
